Structure different types of operations

jlvivero opened the pull request:
this should solve #6 
#4 is no longer necessary with this either

Still need to add some implementations but I want opinions on if the changes are fine

jlvivero committed:
Structure different types of operations

A-deLuna commented on base.c:
```diff
@@ -168,8 +168,27 @@ int isOneByteOp() {
+  if(isVarOp()){
+      return 1;
+  }
+  else{
```
this else will only match with the previous if, I think you ment to use else if on the whole chain

jlvivero replied:
nope, since all the ifs have returns on them it's functionally the same thing

if something return ends the chain, if it's false it will go to the next if, it will only reach the else if all the if's are false

jlvivero commented:
nope, since all the ifs have returns on them it's functionally the same thing

if something  return ends the chain, if it's false it will go to the next if, so every if has functionally an else except the last one

A-deLuna commented on base.c:
```diff
@@ -168,8 +168,27 @@ int isOneByteOp() {
 
+int isOp() {
+  if(isOneByteOp()){
+      return 0;
```
what if instead of returning raw numbers we add some #define statements in the header, and we check against those in the switch statement?

jlvivero replied:
so like `#DEFINE BYTEOP 0` and just do case `BYTEOP:`

jlvivero committed:
Add define for the types of operation and implement said operations

Also added compare to onebyteop since it saves to registry and takes it's parameters from stack

jlvivero committed:
Fix return typo
